pkg/http/request.go

```diff
@@ -0,0 +1,91 @@
+package http
+
+import (
+	"encoding/json"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+)
+
+type (
+	Request struct {
+		*http.Request
+		Body io.Reader
+	}
+
+	BufferedReader struct {
+		buffer []byte
+	}
+)
+
+// NewBufferedReader creates a new reader from readcloser
+func NewBufferedReader(r io.ReadCloser) (b *BufferedReader, err error) {
```

Comment:
You can use a `io.Reader` since you're not closing

Reply:
Wanted to limit it to `io.ReadCloser` as it will be used only for `http.Request` (for now)